brainlit/algorithms/generate_fragments/tests/test_adaptive_thresh.py

```diff
@@ -0,0 +1,223 @@
+    fm_filter = sitk.FastMarchingBaseImageFilter()
+    fm_filter.SetTrialPoints([seed])
+    fm_filter.SetStoppingValue(stopping_value)
+    fm_img = fm_filter.Execute(speed_img)
+    fm_img = sitk.Cast(sitk.RescaleIntensity(fm_img), sitk.sitkUInt8)
+    labels_predicted = sitk.GetArrayFromImage(fm_img)
+    labels_predicted = (~labels_predicted.astype(bool)).astype(int)
+    labels = fast_marching_seg(img, seed, sigma=sigma)
+    np.testing.assert_array_equal(labels, labels_predicted)
+
+
+def test_level_set_seg():
+    Gx = np.array([])
+    for x in range(0, 101):
+        Gx = np.insert(Gx, x, np.exp(-((x - 50) ** 2) / (2 * (2 ** 2))))
+
+    img = np.repeat([Gx], repeats=30, axis=0)
+    seed = (50, 15)
+    lower_threshold = None
+    upper_threshold = None
+    factor = 2
+    max_rms_error = 0.02
+    num_iter = 1000
+    curvature_scaling = 0.5
+    propagation_scaling = 1
+    _, img_T1_255 = get_img_T1(img)
+    seg = sitk.Image(img_T1_255.GetSize(), sitk.sitkUInt8)
+    seg.CopyInformation(img_T1_255)
+    seg[seed] = 1
+    seg = sitk.BinaryDilate(seg, [1] * seg.GetDimension())
+    stats = sitk.LabelStatisticsImageFilter()
+    stats.Execute(img_T1_255, seg)
+
+    if lower_threshold == None:
+        lower_threshold = stats.GetMean(1) - factor * stats.GetSigma(1)
+    if upper_threshold == None:
+        upper_threshold = stats.GetMean(1) + factor * stats.GetSigma(1)
+
+    init_ls = sitk.SignedMaurerDistanceMap(
+        seg, insideIsPositive=True, useImageSpacing=True
+    )
+
+    lsFilter = sitk.ThresholdSegmentationLevelSetImageFilter()
+    lsFilter.SetLowerThreshold(lower_threshold)
+    lsFilter.SetUpperThreshold(upper_threshold)
+    lsFilter.SetMaximumRMSError(max_rms_error)
+    lsFilter.SetNumberOfIterations(num_iter)
+    lsFilter.SetCurvatureScaling(curvature_scaling)
+    lsFilter.SetPropagationScaling(propagation_scaling)
+    lsFilter.ReverseExpansionDirectionOn()
+    ls = lsFilter.Execute(init_ls, sitk.Cast(img_T1_255, sitk.sitkFloat32))
+    labels_predicted = sitk.GetArrayFromImage(ls > 0)
+    labels = level_set_seg(img, seed, lower_threshold=None, upper_threshold=None)
+    np.testing.assert_array_equal(labels, labels_predicted)
+
+
+def test_connected_threshold():
+    # create an image with 4 layers of gray scales
+    G1 = np.full((4, 4), 255)
+    G2 = np.full((4, 4), 200)
+    G3 = np.full((4, 4), 100)
+    G4 = np.full((4, 4), 0)
+    img = np.concatenate((G1, G2, G3, G4)).reshape(4, 4, 4)
+    # seed at the first layer and give a lower_threshold
+    labels = connected_threshold(img, [(0, 0, 0)], lower_threshold=150)
+    labels_predicted = np.concatenate(
+        ((G1 / G1).astype(int), (G2 / G2).astype(int), G3 - G3, G4 - G4)
+    ).reshape(4, 4, 4)
+    np.testing.assert_array_equal(labels, labels_predicted)
+    # seed at the first layer without giving a lower_threshold
+    labels = connected_threshold(img, [(0, 0, 0)])
+    labels_predicted = np.concatenate(
```

Comment:
comment why this is the expected output

Reply:
[Reasons](https://github.com/neurodata/brainlit/blob/CTalgCodCov/brainlit/algorithms/generate_fragments/tests/test_adaptive_thresh.py#L138-L158) are explained.